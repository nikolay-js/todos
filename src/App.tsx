import React, { useReducer, Reducer, useEffect } from 'react';
import ToDos from './components/ToDos/ToDos';
import { ContextApp, initialState } from './context';
import { State, Action, TContextState } from './types';
import tasksReducer from './reducers/tasksReducer';

function App() {
  const [state, changeState] = useReducer<Reducer<State, Action>>(tasksReducer, initialState);

  useEffect(() => {
    localStorage.setItem('todo-list', JSON.stringify(state.tasks))
  }, [state.tasks]);

  const ContextState: TContextState = {
    state,
    changeState
  };

  return (
    <ContextApp.Provider value={ContextState}>
      <ToDos />
    </ContextApp.Provider>
  );
}

export default App;
