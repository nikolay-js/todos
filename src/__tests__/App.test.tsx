import * as React from 'react';
import { shallow } from 'enzyme';

import App from "../App";
import { fireEvent, render, waitFor, cleanup } from "@testing-library/react";
import { act } from "react-dom/test-utils";

describe('<App />', () => {

  afterEach(cleanup);

  it('renders the component without changes', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  it("renders the heading", () => {
    const { container } = render(<App />);
    const h1 = container.querySelector('h1');
    if (h1 !== null) {
      expect(h1.textContent).toEqual('todos');
    };
  });

  it('should render right input value', async () => {
    const { container } = render(<App />);
    const input = container.querySelector('input');
    const button = container.querySelector('button');

    if (input !== null && button !== null) {
      expect(input.getAttribute('value')).toEqual('');
      act(() => {
        fireEvent.change(input, {
          target: {
            value: 'test'
          },
        })
      })
      await waitFor(() => {
        expect(input.getAttribute('value')).toEqual('test');
      });
      act(() => {
        fireEvent.click(button)
      })
      await waitFor(() => {
        expect(input.getAttribute('value')).toEqual('');
      });
    };
  });
})
