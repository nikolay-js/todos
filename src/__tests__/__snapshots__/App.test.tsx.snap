// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the component without changes 1`] = `
<ContextProvider
  value={
    {
      "changeState": [Function],
      "state": {
        "filter": "all",
        "taskText": "",
        "tasks": [],
      },
    }
  }
>
  <ToDos />
</ContextProvider>
`;
